app/routes/circulars.edit.$circularId/CircularEditForm.tsx

```diff
@@ -164,150 +181,214 @@ export function CircularEditForm({
   const changesHaveBeenMade =
     body.trim() !== defaultBody.trim() ||
     subject.trim() !== defaultSubject.trim() ||
-    format !== defaultFormat
+    format !== defaultFormat ||
+    submitter !== updatedSubmitter?.trim() ||
+    Date.parse(`${defaultDate[0]} ${defaultDate[1].substring(0, 5)}`) !==
+      dateNumber
+
+  const userIsModerator = useModStatus()
+
+  useEffect(() => {
+    setDateString(`${date} ${time} UTC`)
+  }, [date, time])
```

Comment:
Why do we need `useEffect` here?

Reply:
the dateString needs to update consistently whenever date or time gets updated, can this be better written?

Reply:
Update, useEffect is needed, since there are multiple state changes that effect the dateString. Without it, I noticed inconsistencies in updates to the field